### PR TITLE
Feature/build image release artefacts 3.0

### PR DESCRIPTION
**Pull request by priyadishah**

### CODE REVIEW

File: scripts/install-lansa-base.ps1 [hunk 1]
```diff
@@ -98,7 +98,7 @@ try
     if ( !(test-path $TempPath) ) {
         New-Item $TempPath -type directory -ErrorAction SilentlyContinue | Out-Default | Write-Host
     }
-
+    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
```
Comment (robe070) [inline scripts/install-lansa-base.ps1]: You need a comment to explain this line.

File: Build Image Release Artefacts 3.0.yml [hunk 2]
```diff
@@ -0,0 +1,1654 @@
+# Variable 'atomic-build' was defined in the Variables tab
+# Variable 'atomic-build-ps' was defined in the Variables tab
+# Variable 'azureUserId' was defined in the Variables tab
```
Comment (robe070) [inline Build Image Release Artefacts 3.0.yml]: azureUserId should be in a variable group - probably already declared.

File: Build Image Release Artefacts 3.0.yml [hunk 3]
```diff
@@ -0,0 +1,1654 @@
...
+# Variable 'Build-w16d-15-0j' was defined in the Variables tab
+# Variable 'Build-w19d-14-2' was defined in the Variables tab
+# Variable 'Build-w19d-14-2j' was defined in the Variables tab
+# Variable 'Build-w19d-15-0' was defined in the Variables tab
+# Variable 'Build-w19d-15-0j' was defined in the Variables tab
+# Variable 'clientsecret' was defined in the Variables tab
+# Variable 'GitBranch' was defined in the Variables tab
+# Variable 'GitBranchPR' was defined in the Variables tab
+# Variable 'GitUserName' was defined in the Variables tab
+# Variable 'JpnLanguage' was defined in the Variables tab
```
Comment (robe070) [inline Build Image Release Artefacts 3.0.yml]: JpnLanguage - Declare in YAML

File: Build Image Release Artefacts 3.0.yml [hunk 4]
```diff
@@ -0,0 +1,1654 @@
...
+# Variable 'KeypairName' was defined in the Variables tab
```
Comment (robe070) [inline Build Image Release Artefacts 3.0.yml]: KeypairName - Declare in Variable Group

File: Build Image Release Artefacts 3.0.yml [hunk 5]
```diff
@@ -0,0 +1,1654 @@
...
+# Variable 'quickstart-scripts' was defined in the Variables tab
+# Variable 'RunWindowsUpdates' was defined in the Variables tab
+# Variable 'TerminateInstance' was defined in the Variables tab
+# Variable 'VersionText-w12r2d-14-2del' was defined in the Variables tab
+# Variable 'VersionText-w12r2d-15-0del' was defined in the Variables tab
+# Variable 'VersionText-w16d-14-2del' was defined in the Variables tab
+# Variable 'VersionText-w16d-15-0del' was defined in the Variables tab
+# Variable 'VersionText-w19d-14-2del' was defined in the Variables tab
+# Variable 'VersionText-w19d-15-0del' was defined in the Variables tab
```
Comment (robe070) [inline Build Image Release Artefacts 3.0.yml]: All the 'del' variables are unused. Delete them.

File: Build Image Release Artefacts 3.0.yml [hunk 6]
```diff
@@ -0,0 +1,1654 @@
...
+# Variable 'Windows-2012-AMI' was defined in the Variables tab
+# Variable 'Windows-2016-AMI' was defined in the Variables tab
+# Variable 'Windows-2016-AMI-JPN' was defined in the Variables tab
+# Variable 'Windows-2019-AMI' was defined in the Variables tab
+# Variable 'Windows-2019-AMI-JPN' was defined in the Variables tab
```
Comment (robe070) [inline Build Image Release Artefacts 3.0.yml]: All the Windows- variables need to be declared in the YAML

**Review by robe070 — CHANGES_REQUESTED** [review]
...and remove all the comments at the start about the variables.

I've not reviewed the YAML in depth,

**Comment by priyadishah** [comment]
all changes are done

File: Build Image Release Artefacts 3.0.yml [hunk 7]
```diff
@@ -0,0 +1,1634 @@
...
+  - name: Windows-2016-AMI-JPN
+    value: Windows_Server-2016-Japanese-Full-Base*
+  - name: Windows-2019-AMI
+    value: Windows_Server-2019-English-Full-SQL_2019_Express*
+  - name: Windows-2019-AMI-JPN
+    value: Windows_Server-2019-Japanese-Full-Base*
+  - name: Windows-2022-AMI
+    value: Windows_Server-2022-English-Full-SQL_2019_Express*
+  - name: Windows-2022-AMI-JPN
+    value: Windows_Server-2022-Japanese-Full-Base*
```
Comment (robe070) [inline Build Image Release Artefacts 3.0.yml]: All the Windows_nnnn variables above are shared with "AWS Images - Cookbooks CI 3.0".
So get Nikhil or Tarun to create a Variable Group called "AWS base AMI Image Names" and add all of these variables to that. Please inform Divinash of the same.
@nkhl-accolite

**Review by robe070 — CHANGES_REQUESTED** [review]

File: Build Image Release Artefacts 3.0.yml (hunk 7 above)
Comment (priyadishah) [reply Build Image Release Artefacts 3.0.yml]: removed windows_nnnn variables from yaml & added in variable group and that is mentioned in yaml file.

Comment (priyadishah) [reply Build Image Release Artefacts 3.0.yml]: i have removed Windows_nnnn variables from yaml file & added in mentioned variable group and mentioned over yaml as well

**Review by robe070 — CHANGES_REQUESTED** [review]
Does Windows 2016 and 2019 still build?
Obviously you will need to remove all the code that you have commented out.

You are not prompting for the variables when you run the pipeline.
Build-All should default to True.

azureUserId and awsCredentials should come from a Variable Group

atomic-build, max_retry, quickstart-scripts, and regionName  should be declared in the YAML.

**Comment by robe070** [comment]
A note for me that above commits have not actioned my comments from the last review.

**Comment by priyadishah** [comment]
> A note for me that above commits have not actioned my comments from the last review.

Hi @robe070 I was working release pipeline so couldn't make above changes. Yes, I made above changes now. 

Does Windows 2016 and 2019 still build? Obviously you will need to remove all the code that you have commented out.
- Yes, Windows 2016, 2019 & 2022 build pipeline work, and removed commented lines.

You are not prompting for the variables when you run the pipeline. Build-All should default to True.
- When you click on Run pipeline, it will prompt for the variables.

azureUserId and awsCredentials should come from a Variable Group. atomic-build, max_retry, quickstart-scripts, and regionName should be declared in the YAML.
- above changes are done.

File: Build Image Release Artefacts.yml [hunk 8]
```diff
@@ -1,12 +1,22 @@
 name: Build Image Release Artefacts 3.0-$(Build.BuildId)
```
Comment (robe070) [inline Build Image Release Artefacts.yml]: Remove 3.0 from the Name property in the YAML.

**Review by robe070 — CHANGES_REQUESTED** [review]
Name of `Build Image Release Artefacts.yml` needs to be `AWS Build Image Release Artefacts.yml`

**Comment by robe070** [comment]
I also see that Win2022 and Win2016 have tested successfully. Win2019 has sysprepped successfully and is building the AMI.
Looking good!

File: Build Image Release Artefacts.yml (hunk 8 above)
Comment (priyadishah) [reply Build Image Release Artefacts.yml]: done, removed 3.0

**Comment by priyadishah** [comment]
> Name of `Build Image Release Artefacts.yml` needs to be `AWS Build Image Release Artefacts.yml`

change done

**Comment by priyadishah** [comment]
> I also see that Win2022 and Win2016 have tested successfully. Win2019 has sysprepped successfully and is building the AMI. Looking good!

correct @robe070 , the code (logic) which you provided for Windows 2016, 2019 & 2022 sysprep, it's working fine.

**Review by robe070 — CHANGES_REQUESTED** [review]
$MaxRetry needs to be passed to `bake-scalable-azure-image-pipeline.ps1`

**Comment by priyadishah** [comment]
> $MaxRetry needs to be passed to `bake-scalable-azure-image-pipeline.ps1`

change is done.

**Comment by robe070** [comment]
There are conflicts. You need to REBASE your changes on my latest branch and fix the conflicts. Then I'll be able to merge the PR.

**Comment by priyadishah** [comment]
> There are conflicts. You need to REBASE your changes on my latest branch and fix the conflicts. Then I'll be able to merge the PR.

I have done sync fork and resolved the conflicts. Please check

File: scripts/install-lansa-base.ps1 [hunk 9]
```diff
@@ -171,7 +170,11 @@ try
                 $downloaded = $true
             } catch {
                 $_
+<<<<<<< HEAD
                 $TotalFailedDownloadAttempts += 1
+=======
+                $TotalFailedDownloadAttempts += 1 
+>>>>>>> debug/paas
```
Comment (robe070) [inline scripts/install-lansa-base.ps1]: This set of changes will not execute! You've left the merge choices in!
And you may need to rebase again if there is a conflict.

**Review by robe070 — CHANGES_REQUESTED** [review]

**Pull request merged**